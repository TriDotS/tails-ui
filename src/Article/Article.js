import PropTypes from 'prop-types'
import React from 'react'

const Article = ({ title, author, description, abstract, footer }) => (
  <div className="w-full">
    <h1 className="font-sans font-thin mb-4">{title}</h1>
    <p className="text-grey mb-3">{author}</p>
    <h2 className="font-sans font-thin leading-normal mb-4">{description}</h2>
    <p className="text-grey-darkest mb-6 leading-tight">{abstract}</p>
    <div>
      {React.cloneElement(footer, {
        className: 'text-black no-underline'
      })}
    </div>
  </div>
)

Article.propTypes = {
  title: PropTypes.string,
  author: PropTypes.string,
  description: PropTypes.string,
  abstract: PropTypes.string,
  footer: PropTypes.node
}

export default Article
