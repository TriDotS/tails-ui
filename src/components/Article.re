let component = ReasonReact.statelessComponent("Article");

let make =
    (
      ~title="Article Title",
      ~author="Article Author",
      ~description="Article Description",
      ~abstract="Article Abstract",
      ~footer=?,
      _children,
    ) => {
  ...component,
  render: _self => {
    let footerComponent =
      switch (footer) {
      | None => ReasonReact.nullElement
      | Some(footer) => ReasonReact.arrayToElement(footer)
      };
    <div className="w-full font-sans">
      <h1 className="font-sans font-thin mb-4">
        (ReasonReact.stringToElement(title))
      </h1>
      <p className="text-grey mb-3">
        (ReasonReact.stringToElement(author))
      </p>
      <h2 className="font-sans font-thin leading-normal mb-4">
        (ReasonReact.stringToElement(description))
      </h2>
      <p className="text-grey-darkest mb-6 leading-tight">
        (ReasonReact.stringToElement(abstract))
      </p>
      <div className="text-black no-underline uppercase">
        footerComponent
      </div>
    </div>;
  },
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~title=jsProps##title,
      ~author=jsProps##author,
      ~description=jsProps##description,
      ~abstract=jsProps##abstract,
      ~footer=?jsProps##footer,
      [||],
    )
  );
