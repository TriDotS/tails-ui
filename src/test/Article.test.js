import Article from '../components/Article'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Article />, div)
})

test('article snapshot', () => {
  const tree = renderer.create(<Article />).toJSON()
  expect(tree).toMatchSnapshot()
})
