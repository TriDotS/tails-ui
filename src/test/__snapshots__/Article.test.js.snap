// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article snapshot 1`] = `
<div
  className="w-full font-sans"
>
  <h1
    className="font-sans font-thin mb-4"
  />
  <p
    className="text-grey mb-3"
  />
  <h2
    className="font-sans font-thin leading-normal mb-4"
  />
  <p
    className="text-grey-darkest mb-6 leading-tight"
  />
</div>
`;
